app/src/main/java/com/example/geoquiz/MainActivity.kt

```diff
@@ -54,46 +65,52 @@ class MainActivity : AppCompatActivity() {
             }
             updateQuestion()
         }
+        previousButton.setOnClickListener {
+            if (currentIndex > 0) {
+                currentIndex = (currentIndex - 1)
+            }
+            updateQuestion()
+        }
         updateQuestion()
     }
 
     private fun updateQuestion() {
         val questionTextResId = questionBank[currentIndex].textResId
         questionTextView.setText(questionTextResId)
+        answeredQuestionCheck()
     }
 
     private fun checkAnswer(userAnswer: Boolean) {
         val answer = questionBank[currentIndex].answer
-        val messageResId = if (answer == userAnswer) {
-            R.string.correct
+        var messageResId = 0
+        if (answer == userAnswer) {
+            messageResId = R.string.correct
+            correctAnswers++
         } else {
-            R.string.incorrect
+            messageResId = R.string.incorrect
         }
         Toast.makeText(this, messageResId, Toast.LENGTH_SHORT).show()
     }
 
-    override fun onStart() {
-        super.onStart()
-        Log.d(TAG, "onStart called ")
-    }
 
-    override fun onResume() {
-        super.onResume()
-        Log.d(TAG, "onResume called ")
-    }
-
-    override fun onPause() {
-        super.onPause()
-        Log.d(TAG, "onPause called ")
-    }
-
-    override fun onStop() {
-        super.onStop()
-        Log.d(TAG, "onStop called ")
+    private fun answeredQuestionCheck() {
```

Comment:
Based on the name I assume the function to check all answered questions.
Instead, it sets the button states based on the current question's answered state.
I would rename to `updateQuestionButtonsState(isCurrentQuestionAnswered: Boolean)` - and pass `questionBank[currentIndex].answeredQuestion` as param